src/main/java/games/strategy/ui/SwingComponents.java

```diff
@@ -59,8 +56,111 @@
 import games.strategy.net.OpenFileUtility;
 import games.strategy.triplea.UrlConstants;
 
+/**
+ * Wrapper/utility class to give Swing components a nicer API. This class is to help extract pure UI code out of
+ * the rest of the code base. This also gives us a cleaner interface between UI and the rest of the code.
+ */
 public class SwingComponents {
   private static final String PERIOD = ".";
+  private static final Collection<String> visiblePrompts = new HashSet<>();
+
+  /**
+   * Convenience method to create {@code AbstractAction} instances using lambdas.
+   * Example usage:
+   * <code><pre>
+   *   AbstractAction action = SwingComponents.newAbstractionAction(() -> {
+   *    if (value) {
+   *      doSomething();
+   *    }
+   *   });
+   * </pre></code>
+   */
+  public static AbstractAction newAbstractAction(final Runnable action) {
+    return new AbstractAction() {
+      private static final long serialVersionUID = -280371946771796597L;
+
+      @Override
+      public void actionPerformed(final ActionEvent e) {
+        action.run();
+      }
+    };
+  }
+
+  public enum KeyDownMask {
+    META_DOWN(InputEvent.META_DOWN_MASK),
+    CTRL_DOWN(InputEvent.CTRL_DOWN_MASK);
+
+    private final int code;
+
+    KeyDownMask(final int code) {
+      this.code = code;
+    }
+  }
+
+  public static void addSpaceKeyListener(final JComponent component, final Runnable runnable) {
+    addKeyListener(component, KeyStroke.getKeyStroke(KeyEvent.VK_SPACE, 0), runnable);
+  }
+
+  public static void addEnterKeyListener(final JComponent component, final Runnable runnable) {
+    addKeyListener(component, KeyStroke.getKeyStroke(KeyEvent.VK_ENTER, 0), runnable);
+  }
+
+  public static void addEscapeKeyListener(
+      final RootPaneContainer dialog,
+      final Runnable keyDownAction) {
+    if(dialog.getRootPane() != null) {
```

Comment:
Checkstyle: "WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)"